File: Sources/Api/PasswordlessLoginAPI.m
//
//  PasswordlessLoginAPI.m
//  LoginRadiusSDK
//
//  Created by LoginRadius on 30/07/18.
//

#import "PasswordlessLoginAPI.h"

@implementation PasswordlessLoginAPI
+ (instancetype)passwordlessInstance{
    static dispatch_once_t onceToken;
    static PasswordlessLoginAPI *instance;
    
    dispatch_once(&onceToken, ^{
        instance = [[PasswordlessLoginAPI alloc] init];
    });
    
    return instance;
}

-(void)passwordlessLoginWithEmail:(NSString *)email passwordlesslogintemplate:(NSString *)passwordlesslogintemplate verificationurl:(NSString *)verificationurl completionHandler:(LRAPIResponseHandler)completion {
    NSString *passwordlesslogin_template = passwordlesslogintemplate ? passwordlesslogintemplate: @"";
    [[LoginRadiusREST apiInstance] sendGET:@"identity/v2/auth/login/passwordlesslogin/email"
                               queryParams:@{
                                             @"apikey": [LoginRadiusSDK apiKey],
                                             @"email": email,
                                             @"passwordlesslogintemplate": passwordlesslogin_template,
                                             @"verificationurl":[LoginRadiusSDK verificationUrl]
                                             }
                         completionHandler:completion];
}
    
-(void)passwordlessLoginWithUserName:(NSString *)username passwordlesslogintemplate:(NSString *)passwordlesslogintemplate verificationurl:(NSString *)verificationurl completionHandler:(LRAPIResponseHandler)completion{
    NSString *passwordlesslogin_template = passwordlesslogintemplate ? passwordlesslogintemplate: @"";
    [[LoginRadiusREST apiInstance] sendGET:@"identity/v2/auth/login/passwordlesslogin/email"
                               queryParams:@{
                                             @"apikey": [LoginRadiusSDK apiKey],
                                             @"username": username,
                                             @"passwordlesslogintemplate": passwordlesslogin_template,
                                             @"verificationurl":[LoginRadiusSDK verificationUrl]
                                             }
                         completionHandler:completion];
    
}

-(void)passwordlessLoginVerificationWithVerificationToken:(NSString *)verificationtoken welcomeemailtemplate:(NSString *)welcomeemailtemplate completionHandler:(LRAPIResponseHandler)completion {
    NSString *welcomeemail_template = welcomeemailtemplate ? welcomeemailtemplate: @"";
    [[LoginRadiusREST apiInstance] sendGET:@"identity/v2/auth/login/passwordlesslogin/email/verify"
                               queryParams:@{
                                             @"apikey": [LoginRadiusSDK apiKey],
                                             @"verificationtoken": verificationtoken,
                                             @"welcomeemailtemplate": welcomeemail_template
                                             }
                         completionHandler:completion];
    
}


-(void)passwordlessLoginSendOtpWithPhone:(NSString *)phone smstemplate:(NSString *)smstemplate completionHandler:(LRAPIResponseHandler)completion{
    NSString *sms_template = smstemplate ? smstemplate: @"";
    [[LoginRadiusREST apiInstance] sendGET:@"identity/v2/auth/login/passwordlesslogin/otp"
                               queryParams:@{
                                             @"apikey": [LoginRadiusSDK apiKey],
                                             @"phone": phone,
                                             @"smstemplate": sms_template
                                             }
                         completionHandler:completion];
    
}

-(void)passwordlessPhoneLoginWithPayload:(NSDictionary *)payload smstemplate:(NSString *)smstemplate completionHandler:(LRAPIResponseHandler)completion{
    NSString *sms_template = smstemplate ? smstemplate: @"";
    [[LoginRadiusREST apiInstance] sendPUT:@"identity/v2/auth/login/passwordlesslogin/otp/verify"
                               queryParams:@{
                                             @"apikey": [LoginRadiusSDK apiKey],
                                             @"smstemplate": sms_template
                                             }
                                      body:payload
                         completionHandler:completion];
}

@end

File: Sources/PasswordlessLoginAPI.swift
//
//  PasswordlessLoginAPI.swift
//  LoginRadiusSwift SDK
//
//  Created by Megha Agarwal.
//  Copyright © 2023 LoginRadius Inc. All rights reserved.
//

import Foundation


public class PasswordlessLoginAPI {
    
    public static let shared = PasswordlessLoginAPI()
    
    public init() {}
    
    public func passwordlessLoginWithEmail(email: String, passwordlesslogintemplate: String?, verificationurl: String, completionHandler: @escaping LRAPIResponseHandler) {
        let passwordlesslogin_template = passwordlesslogintemplate ?? ""
        LoginRadiusREST.apiInstance.sendGET(url: "identity/v2/auth/login/passwordlesslogin/email", queryParams: [
            "apikey": LoginRadiusSDK.sharedInstance.apiKey,
            "email": email,
            "passwordlesslogintemplate": passwordlesslogin_template,
            "verificationurl": LoginRadiusSDK.sharedInstance.verificationUrl
        ], completionHandler: completionHandler)
    }
    
    public func passwordlessLoginWithUserName(username: String, passwordlesslogintemplate: String?, verificationurl: String, completionHandler: @escaping LRAPIResponseHandler) {
        let passwordlesslogin_template = passwordlesslogintemplate ?? ""
        LoginRadiusREST.apiInstance.sendGET(url: "identity/v2/auth/login/passwordlesslogin/email", queryParams: [
            "apikey": LoginRadiusSDK.sharedInstance.apiKey,
            "username": username,
            "passwordlesslogintemplate": passwordlesslogin_template,
            "verificationurl": LoginRadiusSDK.sharedInstance.verificationUrl
        ], completionHandler: completionHandler)
    }
    
    public func passwordlessLoginVerificationWithVerificationToken(verificationtoken: String, welcomeemailtemplate: String?, completionHandler: @escaping LRAPIResponseHandler) {
        let welcomeemail_template = welcomeemailtemplate ?? ""
        LoginRadiusREST.apiInstance.sendGET(url: "identity/v2/auth/login/passwordlesslogin/email/verify", queryParams: [
            "apikey": LoginRadiusSDK.sharedInstance.apiKey,
            "verificationtoken": verificationtoken,
            "welcomeemailtemplate": welcomeemail_template
        ], completionHandler: completionHandler)
    }
    
    public func passwordlessLoginSendOtpWithPhone(phone: String, smstemplate: String?, completionHandler: @escaping LRAPIResponseHandler) {
        let sms_template = smstemplate ?? ""
        LoginRadiusREST.apiInstance.sendGET(url: "identity/v2/auth/login/passwordlesslogin/otp", queryParams: [
            "apikey": LoginRadiusSDK.sharedInstance.apiKey,
            "phone": phone,
            "smstemplate": sms_template
        ], completionHandler: completionHandler)
    }
    
    public func passwordlessPhoneLoginWithPayload(payload: [String: Any], smstemplate: String?, completionHandler: @escaping LRAPIResponseHandler) {
        let sms_template = smstemplate ?? ""
        LoginRadiusREST.apiInstance.sendPUT(url: "identity/v2/auth/login/passwordlesslogin/otp/verify", queryParams: [
            "apikey": LoginRadiusSDK.sharedInstance.apiKey,
            "smstemplate": sms_template
        ], body: payload, completionHandler: completionHandler)
    }
    
}
